euphoria-examples/src/main/java/cz/seznam/euphoria/examples/HFileLoader.java

```diff
@@ -0,0 +1,170 @@
+import cz.seznam.euphoria.spark.SparkExecutor;
+import java.util.stream.Stream;
+
+/**
+ * Load data to HBase from various sources and in various bulks.
+ */
+public class HFileLoader {
+
+  public static void main(String[] args) throws URISyntaxException, IOException {
+    Preconditions.checkArgument(
+        args.length >= 3, "Please specify <input_uri> <output_uri> <executor> [<tmp_dir>]");
+
+    URI input = new URI(args[0]);
+    URI output = new URI(args[1]);
+    Executor executor = Executors.createExecutor(
+        args[2],
+        KeyValue.class);
+    String tmp = "/tmp/hfileloader";
+    if (args.length > 3) {
+      tmp = args[2];
+    }
+    HFileLoader app = new HFileLoader(input, output, tmp, executor);
+    app.run();
+  }
+
+  private final Configuration conf = HBaseConfiguration.create();
+  private final String table;
+  private final String outputQuorum;
+  private final String tmpDir;
+  private final URI input;
+  private final URI output;
+  private final Executor executor;
+  private final Flow flow;
+
+  HFileLoader(URI input, URI output, String tmpDir, Executor executor) {
+    this.table = getPath(output);
+    this.outputQuorum = output.getAuthority();
+    this.tmpDir = tmpDir;
+    this.input = input;
+    this.output = output;
+    this.executor = executor;
+    this.flow = Flow.create();
+  }
+
+  @SuppressWarnings("unchecked")
+  private void run() {
+    final Dataset<Cell> ds;
+    flow.getSettings().setInt(
+        "euphoria.flink.batch.list-storage.max-memory-elements", 100);
+    switch (input.getScheme()) {
+      case "kafka":
+      {
+        Settings settings = new Settings();
+        settings.setInt(KafkaSource.CFG_RESET_OFFSET_TIMESTAMP_MILLIS, 0);
+        Dataset<Pair<byte[], byte[]>> raw = flow.createInput(
+            new KafkaSource(input.getAuthority(),
+                getPath(input),
+                settings));
+        ds = MapElements.of(raw)
+            .using(p -> toCell(p.getSecond()))
+            .output();
+        break;
+      }
+      case "hdfs":
+      case "file":
+      {
+        Dataset<Pair<ImmutableBytesWritable, Cell>> raw = flow.createInput(
+            new SequenceFileSource<>(ImmutableBytesWritable.class,
+                (Class) KeyValue.class, input.toString()));
+        ds = MapElements.of(raw)
+            .using(Pair::getSecond)
+            .output();
+        break;
+      }
+      case "hbase":
+      {
+        HBaseSource source = getHBaseSource(input, conf);
+        Dataset<Pair<ImmutableBytesWritable, Result>> raw = flow.createInput(
+            source);
+
+        Dataset<Cell> tmp = FlatMap.of(raw)
+            .using(ResultUtil.toCells())
+            .output();
+
+        if (executor instanceof SparkExecutor) {
+          tmp = ReduceByKey.of(tmp)
+              .keyBy(Object::hashCode)
+              .reduceBy((Stream<Cell> s, Collector<Cell> ctx) -> s.forEach(ctx::collect))
+              .outputValues();
+        }
```

Comment:
not cool, it should at least have a good comments with explanation

Reply:
Added comment.